lib/ansible/module_utils/facts.py

```diff
@@ -3067,15 +3070,33 @@ def get_file_lines(path):
     return ret
 
 def ansible_facts(module):
+    # Retrieve module parameters
+    gather_subset = [ 'all' ]
+    if 'gather_subset' in module.params:
+        gather_subset = module.params['gather_subset']
```

Comment:
These three lines can now be:
```
gather_subset = module_params['gather_subset']
```

because the argument_spec for gather_subset ensures that we'll always have a value for this .